### PR TITLE
Fix issue with native projects .vcxproj being built in IDE not seeing…

### PR DESCRIPTION
**Pull request by bbowman**
… that they had the Packaging project installed.

Wasn't sure how to go about unit testing since Design time is involved.

  === TEST EXECUTION SUMMARY ===
     NuGet.Build.Packaging.Tests  Total: 119, Errors: 0, Failed: 8, Skipped: 0, Time: 13.841s

Are failures expected?

### CODE REVIEW

File: src/Build/NuGet.Build.Packaging.Tasks/NuGet.Build.Packaging.targets [hunk 1]
```diff
@@ -63,6 +63,11 @@ Copyright (c) .NET Foundation. All rights reserved.
 			<IsPackable>$(IsPackable)</IsPackable>
 			<IsNuGetized>$(IsNuGetized)</IsNuGetized>
 		</TargetPathWithTargetPlatformMoniker>
+		<!-- Design Time is annoying and uses a different output item group to return this. Also adust that -->
+		<_MainProjectOutput>
```
Comment (kzu) [inline src/Build/NuGet.Build.Packaging.Tasks/NuGet.Build.Packaging.targets:67]: If this is specific to VC++, it should go into their targets, not nugetizer

**Comment by kzu** [comment]
@bbowman the CI build is green with all tests, so it must be some local issue. See https://ci.appveyor.com/project/MobileEssentials/nuget-build-packaging/build/0.1.163-pr61/tests

File: src/Build/NuGet.Build.Packaging.Tasks/NuGet.Build.Packaging.targets (hunk 1 above)
Comment (bbowman) [reply src/Build/NuGet.Build.Packaging.Tasks/NuGet.Build.Packaging.targets:67]: I agree in principal but it seems a little harder / less expidient to get them on the same page. You can see the issue at: C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140

 or wherever you've installed VS. They override GetTargetPathWithTargetPlatformMoniker target to use a different itemGroup. 

The nugetizer solution here relies on a standardized item type as the return type from that target but that may not always be the case. Doesn't seem like the end of the world to do this here?

Comment (kzu) [reply src/Build/NuGet.Build.Packaging.Tasks/NuGet.Build.Packaging.targets:67]: It's the item type in the MS.Common targets, so I'd say the VC++ targets need to not-break them by redefining them in an incompatible way.

In the meantime, you can always re-declare the broken target in your own targets and make it compatible again with the MS.Common targets.

It would be good to get the VC++ guys in the loop. Dunno who to @ though

Comment (bbowman) [reply src/Build/NuGet.Build.Packaging.Tasks/NuGet.Build.Packaging.targets:67]: @rainersigwald or someone on the msbuild project may know more?

Comment (kzu) [reply src/Build/NuGet.Build.Packaging.Tasks/NuGet.Build.Packaging.targets:67]: Not really, I think, since it's a VC++ targets specific issue, rather than an MSBuild one.

Comment (rainersigwald) [reply src/Build/NuGet.Build.Packaging.Tasks/NuGet.Build.Packaging.targets:67]: @olgaark @yuehuang010?

**Comment by olgaark** [comment]
Thanks Rainer for including us.

Can somebody please explain what the issue actually is?

For the GetTargetPathWithTargetPlatformMoniker target: we are overriding it because at some point we were asked to do so as we are overriding GetTargetPath. If we don’t need to do it anymore, we’ll be happy to remove it. But native outputs are way more complicated than managed ones, so we still might need it. We can definitely modify it if this makes life easier, so let us know.

Thanks,
Olga

**Comment by bbowman** [comment]
@olgaark @yuehuang010 @rainersigwald 

The issue we are seeing is that the native design time targets, Microsoft.Cpp.DesignTime.targets, uses **_MainProjectOutput** as the ItemGroup returned by the **GetTargetPathWithTargetPlatformMoniker** target. This is an issue for Nugetizer because it assumes that **TargetPathWithTargetPlatformMoniker** will be used and adjusts its ItemDefinitionGroup as a way to smuggle extra information out of the standard targets. 

The _real_ way to accomplish what @kzu wants here is a way to interrogate project files for the targets they define but this is obviously complicated by Conditions and the rest (not easily observable from outside the build). Instead he sets a flag on a standard target to know that he is safe to call his target on a referenced project.

His approach _should_ be safe if everyone obeys the contract laid out by the common props and targets but there is a small violation for DesignTime which bit me when building in visual studio.

**Comment by bbowman** [comment]
tl;dr

Please update Microsoft.Cpp.DesignTime.targets to use TargetPathWithTargetPlatformMoniker instead of _MainProjectOutput

**Comment by olgaark** [comment]
OK, should be simple. Can you please open a bug for us (so you’ll know when we fix it)?
You can create a copy of
https://devdiv.visualstudio.com/web/wi.aspx?pcguid=011b8bdf-6d56-4f87-be0d-0092136884d9&id=273357
to get the right path, etc

Thanks,
Olga

**Comment by bbowman** [comment]
Bug: Update Microsoft.Cpp.DesignTime.targets to use TargetPathWithTargetPlatformMoniker instead of _MainProjectOutput 

created. Thanks all!

@kzu I still think there is a useful change here to use GetTargetPathWithTargetPlatformMoniker instead of GetTargetPath. GetTargetPath is overridden elsewhere to return $(TargetPath) etc that may not have the TPM metadata attached. We should be specific in calling the target that we know will have the nugetizer metadata on it. Should I resubmit a PR with just that change and then adjust the _MainProjectOutput in my own targets/props until they fix it?

**Comment by kzu** [comment]
I originally did it that way because that target was introduced in vs2015,
but since that's the minimum supported version anyway, I think it would be
fine to use it to get target path.

**Comment by kzu** [comment]
I believe this should have implemented the suggestion: https://github.com/NuGet/NuGet.Build.Packaging/commit/28e9e699f6980e44984c46a35692a13b8af7907c
